### PR TITLE
Fully connects application e2e from ios/android application to server an...

### PR DESCRIPTION
**Pull request by kranrao**
...d S3/DB.

User sessions:
- Adds CurrentUser service to store sessionIDs
- Adds CurrentUser service to index.html
- Captures user session on user signup and login
- Adds sessionIDs to server request headers

Dataservice functionality:
- Removes mock data
- Removes addMoment method and utilizes momentID
- Connects getMementos method with server
- Adjusts saveMoment to appropriately handle server response
- Adjusts memento controller to appropriately handle server response
- Adjusts memento.create controller to appropriately handle server reponse
- Adjusts mementos controller to appropriately handle server

Downloads:
Adds download service to core module

Memento view:
Adjusts memento.html to render moment items by utilizing item directive functionality

Client hooks:
- Modifies existing client hooks in repository

### CODE REVIEW

File: client/www/app/core/CurrentUser.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      set: function(obj) {
+        user = angular.copy(obj);
+      },
+      get: function() {
+        return angular.copy(user);
+      }
+    };
+  }
+
+})();
```
Comment (omarduarte) [inline client/www/app/core/CurrentUser.js]: Please include a new line!

File: client/www/app/core/dataservice.js [hunk 2]
```diff
@@ -188,96 +226,34 @@
...
+        })
+        .error(function(error){
+          console.error('There was an error saving memento');
+          throw error;
+        });
     }
 
   }
-
 })();
```
Comment (omarduarte) [inline client/www/app/core/dataservice.js]: new line

File: client/www/app/core/download.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        return data;
+      })
+      .error(function(data, status, headers, config) {
+        console.log('There was an error downloading moment item from S3');
+      });
+
+    }
+  }
+
+})();
```
Comment (omarduarte) [inline client/www/app/core/download.js]: new line

**Comment by kranrao** [comment]
Omar - i fixed these issues and included an extra space for all other files that did not have these

**Pull request merged**